dev-docs/e2e/bare-metal-runner.md

```diff
@@ -1,5 +1,221 @@
+First install the `mainline` package:
+```bash
+sudo add-apt-repository ppa:cappelikan/ppa
+sudo apt-get update
+sudo apt-get install mainline pkexec
+```
+Now list all available kernel versions and install the latest LTS version. To
+figure out what the latest long term version is, refer to https://kernel.org/.
+```bash
+mainline list
+mainline install <latest LTS version e.g. 6.12.17>
+```
+Reboot the machine to boot automatically into the latest kernel and delete all old ones.
+```bash
+reboot
+mainline uninstall-old
+```
+
+Check that SEV-SNP is enabled. If it's not then it likely needs to be
+enabled in the BIOS. For those steps, either have a look in our docs
+https://docs.edgeless.systems/contrast/getting-started/bare-metal or
+google for "enable AMD SEV in BIOS." Sadly, AMD changes their document
+links from time to time, so we don't link it here.
+
+Once it's enabled, verify this as follows:
+```bash session
+root@hetzner-ax162-snp ~ # journalctl -k -b 0 | grep -i sev
+Feb 27 19:32:31 hetzner-ax162-snp kernel: SEV-SNP: RMP table physical range [0x0000000035500000 - 0x0000000075afffff]
+Feb 27 19:32:31 hetzner-ax162-snp kernel: SEV-SNP: Reserving start/end of RMP table on a 2MB boundary [0x0000000035400000]
+Feb 27 19:32:31 hetzner-ax162-snp kernel: SEV-SNP: Reserving start/end of RMP table on a 2MB boundary [0x0000000075a00000]
+Feb 27 19:32:32 hetzner-ax162-snp kernel: ccp 0000:09:00.5: sev enabled
+Feb 27 19:33:13 hetzner-ax162-snp kernel: ccp 0000:09:00.5: SEV API:1.55 build:32
+Feb 27 19:33:13 hetzner-ax162-snp kernel: ccp 0000:09:00.5: SEV-SNP API:1.55 build:32
+Feb 27 19:33:13 hetzner-ax162-snp kernel: kvm_amd: SEV enabled (ASIDs 100 - 1006)
+Feb 27 19:33:13 hetzner-ax162-snp kernel: kvm_amd: SEV-ES enabled (ASIDs 1 - 99)
+Feb 27 19:33:13 hetzner-ax162-snp kernel: kvm_amd: SEV-SNP enabled (ASIDs 1 - 99)
+```
+
+Install `docker` so that the docker login step in the CI succeeds.
+On Ubuntu, add it to the apt repositories (see https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository).
+```bash
+# Add Docker's official GPG key:
+sudo apt-get update
+sudo apt-get install ca-certificates curl
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+
+# Download docker package
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+```
+
+Install k9s (optional)
+```bash
+curl -fsL https://github.com/derailed/k9s/releases/latest/download/k9s_Linux_amd64.tar.gz | tar -C /usr/local/bin -xz k9s
+```
```

Comment:
nit: I don't think we should recommend this as something to install *on* the runner. But if you see an advantage in doing so, feel free to keep it.

Reply:
removed